eclair-node/src/main/scala/fr/acinq/eclair/api/directives/RouteFormat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.api.directives
+
+import fr.acinq.eclair.router.Router.RouteResponse
+
+// @formatter:off
+sealed trait RouteFormat
+case object NodeIdRouteFormat extends RouteFormat
+case object ShortChannelIdRouteFormat extends RouteFormat
+// @formatter:on
+
+object RouteFormat {
+
+  val NODE_ID = "nodeId"
+  val SHORT_CHANNEL_ID = "shortChannelId"
+
+  def fromString(s: String): RouteFormat = s match {
+    case NODE_ID => NodeIdRouteFormat
+    case SHORT_CHANNEL_ID => ShortChannelIdRouteFormat
+    case _ => throw new IllegalArgumentException(s"invalid route format, possible values are ($NODE_ID, $SHORT_CHANNEL_ID)")
+  }
+
+  def format(route: RouteResponse, format_opt: Option[RouteFormat]): Seq[String] = format(route, format_opt.getOrElse(NodeIdRouteFormat))
```

Comment:
nit: linked to the previous comment, I think the format shouldn't be an option here, the caller should ensure it always provides a format

Reply:
But non-optional format field will make the API backwards incompatible.